### ISSUE TITLE
HaveReturnType() does not support generic types

### ISSUE THREAD

**Issue by pitchart** [TNG/ArchUnitNET#141]
Hi,

I encountered an issue using the `HaveReturnType("...")` check.
Methods that return generic types fail to match.
Used example : 
```
MethodMembers().That().AreDeclaredIn(Controllers())
    .And().ArePublic()
    .And().AreNoConstructors()
    .Should().HaveReturnType(typeof(Task<IActionResult>))
    .Check(Architecture);
```
If I replace usage with `.HaveReturnType(typeof(Task<>))`, it woks fine, but not usefull in this case :-)

**Comment by brandhuf** [comment]
PR #144 should fix this

**Comment by fgather** [comment]
Released as 0.9.1
